streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -320,6 +322,25 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
+  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
+    StreamEffect[Any, Nothing, Byte] {
+      Managed.effectTotal {
+        def pull(): Byte = {
+          val a = is.read()
```

Comment:
this can throw an `IOException`, why not `StreamEffect[Any, IOException, Byte]` then?

Reply:
Yes, good point. We need to try/catch the `IOException` and re-throw it using in `StreamEffect.fail`.